sdk/identity/Azure.Identity/perf/CacheReadPerf.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Test.Perf;
+
+namespace Azure.Template.Perf
+{
+    public class CacheReadPerf : PerfTest<CountOptions>
+    {
+        private ClientSecretCredential _credential;
+        private readonly TokenRequestContext _tokenRequestContext = new(new[] { "https://storage.azure.com/.default" });
+
+        public CacheReadPerf(CountOptions options) : base(options)
+        {
+            _credential = new ClientSecretCredential(
```

Comment:
We use the TestEnvironment in other tests, any reason not to use it?

Reply:
No reason other than I just needed to read the environment without any of the other fanciness.

Reply:
Is it intended to eventually run live in the perf automation?

Reply:
I assume so - without recordings though, what am I missing from TestEnvironment?

Reply:
`test-resource.json.env` support and well known variable names that are created by `New-TestResources`.

Reply:
These tests don't require any resource creation, but point taken. I'll refactor it.

Reply:
Service principal is a "resource" of a kind